services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java

```diff
@@ -161,6 +163,33 @@ public void getCatalogs(FlightProducer.ServerStreamListener listener, BufferAllo
     workerProvider.get().submitWork(runExternalId, userSession, responseHandler, userRequest, TerminationListenerRegistry.NOOP);
   }
 
+  /**
+   * Retrieve the table types and sends the response to given ServerStreamListener.
+   *
+   * @param listener    ServerStreamListener listening to the job result.
+   * @param allocator   BufferAllocator used to allocate the response VectorSchemaRoot.
+   */
+  public void runGetTablesTypes(FlightProducer.ServerStreamListener listener,
+                                BufferAllocator allocator) {
+    try (VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_TABLE_TYPES_SCHEMA,
+      allocator)) {
+      listener.start(vectorSchemaRoot);
+
+      vectorSchemaRoot.allocateNew();
+      VarCharVector tableTypeVector = (VarCharVector) vectorSchemaRoot.getVector("table_type");
+
+      final ImmutableList<TableType> tableTypes = ImmutableList.of(TableType.TABLE, TableType.SYSTEM_TABLE, TableType.VIEW);
```

Comment:
This now has the issue of not adjusting if the enum itself is added to - so this won't work. I would rather we take all the values and filter out UNKNOWN or UNRECOGNIZED.

Reply:
Fixed